### PR TITLE
connect: Strip leading whitespaces in test data.

### LINKED ISSUE #810: connect: README incorrectly describes test data
From the README of the "connect" exercise (emphasis mine):

> The boards look like this (with spaces added for readability, **which won't be in
> the representation passed to your code**):
> 
> ```text
> . O . X .
>  . X X O .
>   O O O X .
>    . X O X O
>     X O O O X
> ```

But if you look at the representation passed to your code:

```ruby
    board = [
      'X O . .',
      ' O X X X',
      '  O X O .',
      '   . O X .',
      '    X X O O'
    ]
```

### PR DESCRIPTION
**Pull request by tryantwit**
README and test show test data as such for readability, test data should
should be passed in like the README indicates, without the spaces added
for readability.

Fixes #810 outstanding issue by striping out leading spaces in constructed board array.

### CODE REVIEW

**Comment by Insti** [comment]
How about also adding the re-generated test suite?

**Comment by Insti** [comment]
Thanks for doing this @tryantwit

File: exercises/connect/.meta/generator/connect_case.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ def workload
     [
       'board = [',
       indent_by(2,board.map(&method(:single_quote)).join(",\n")),
-      ']',
+      "].map {|string| string.gsub(/^ */, '')}",
```
[the remark this answers is not in the document]
Comment (tryantwit) [reply exercises/connect/.meta/generator/connect_case.rb]: Haha yeah, you really are. Can change that easily though.

**Comment by tryantwit** [comment]
If everything looks good, I can squash all this into 1 commit before merge if you want.

**Comment by Insti** [comment]
Yep looks good.
If you want to squash it, make it 2 commits, one for the generator change and one for the new test case file.

**Comment by tryantwit** [comment]
Cleaned up the commits, should be good to go.

**Comment by Insti** [comment]
Awesome, thanks a lot @tryantwit :heart:

**Pull request merged**